app/feature/feature-help-center/src/main/kotlin/com/hedvig/android/feature/help/center/question/HelpCenterQuestionDestination.kt

```diff
@@ -36,161 +35,155 @@ import com.hedvig.android.design.system.hedvig.Surface
 import com.hedvig.android.design.system.hedvig.TopAppBarWithBack
 import com.hedvig.android.design.system.hedvig.plus
 import com.hedvig.android.feature.help.center.ShowNavigateToInboxViewModel
-import com.hedvig.android.feature.help.center.model.Question
+import com.hedvig.android.feature.help.center.data.FAQItem
 import com.hedvig.android.feature.help.center.ui.HelpCenterSection
-import com.hedvig.android.feature.help.center.ui.HelpCenterSectionWithClickableRows
 import com.hedvig.android.feature.help.center.ui.StillNeedHelpSection
 import hedvig.resources.R
 
 @Composable
 internal fun HelpCenterQuestionDestination(
   showNavigateToInboxViewModel: ShowNavigateToInboxViewModel,
-  questionId: Question,
-  onNavigateToQuestion: (questionId: Question) -> Unit,
+  helpCenterQuestionViewModel: HelpCenterQuestionViewModel,
   onNavigateToInbox: () -> Unit,
   onNavigateToNewConversation: () -> Unit,
   onNavigateUp: () -> Unit,
   onNavigateBack: () -> Unit,
 ) {
-  val question = Question.entries.find { it == questionId }
-  val relatedQuestions = question
-    ?.relatedQuestionIds
-    ?.mapNotNull { id ->
-      Question.entries.firstOrNull { it == id }
-    } ?: listOf()
-  HelpCenterQuestionScreen(
-    question = question,
-    relatedQuestions = relatedQuestions,
-    showNavigateToInboxButton = showNavigateToInboxViewModel.uiState.collectAsStateWithLifecycle().value,
-    onNavigateToQuestion = onNavigateToQuestion,
-    onNavigateUp = onNavigateUp,
-    onNavigateBack = onNavigateBack,
-    onNavigateToInbox = onNavigateToInbox,
-    onNavigateToNewConversation = onNavigateToNewConversation,
-  )
+  val uiState by helpCenterQuestionViewModel.uiState.collectAsStateWithLifecycle()
+  when (val state = uiState) {
+    HelpCenterQuestionUiState.Failure -> {
+      FailureScreen(
+        onNavigateBack = onNavigateBack,
+        reload = {
+          helpCenterQuestionViewModel.emit(HelpCenterQuestionEvent.Reload)
+        },
+      )
+    }
+
+    HelpCenterQuestionUiState.Loading -> HedvigFullScreenCenterAlignedProgress()
+    is HelpCenterQuestionUiState.Success -> {
+      HelpCenterQuestionScreen(
+        faqItem = state.faqItem,
+        showNavigateToInboxButton = showNavigateToInboxViewModel.uiState.collectAsStateWithLifecycle().value,
+        onNavigateUp = onNavigateUp,
+        onNavigateToInbox = onNavigateToInbox,
+        onNavigateToNewConversation = onNavigateToNewConversation,
+      )
+    }
+  }
+}
+
+@Composable
+private fun FailureScreen(onNavigateBack: () -> Unit, reload: () -> Unit) {
+  Surface(color = HedvigTheme.colorScheme.backgroundPrimary) {
+    Column(Modifier.fillMaxSize()) {
+      TopAppBarWithBack(
```

Comment:
Is there any of the when brances here which do *not* want to have a top app bar? If no, then we should just also move it up like the surface and the when branches can deal with the rest